designer/custom-http-service-api/src/main/scala/pl/touk/nussknacker/ui/customhttpservice/CustomHttpServiceProviderFactory.scala

```diff
@@ -0,0 +1,16 @@
+package pl.touk.nussknacker.ui.customhttpservice
+
+import cats.effect.{IO, Resource}
+import com.typesafe.config.Config
+import pl.touk.nussknacker.ui.customhttpservice.services.NussknackerServicesForCustomHttpService
+
+import scala.concurrent.ExecutionContext
+
+trait CustomHttpServiceProviderFactory {
+
+  def create(
+      config: Config,
+      services: NussknackerServicesForCustomHttpService
+  )(implicit executionContext: ExecutionContext): Resource[IO, CustomHttpServiceProvider]
```

Comment:
It looks weird that we have IO and we have to pass an `ExecutionContext`. If the `ScenarioService` interface will be IO effect based, would we need it?

Reply:
Interface changed to generic Async :) + removed executionContext.